src/ui.data-table/UTable.vue

```diff
@@ -607,6 +607,18 @@ function getDateSeparatorLabel(separatorDate) {
     : separatorDate;
 }
 
+function getDateSeparatorRowAttrs(rowIndex) {
+  const isPreviousRowChecked = tableRows.value[rowIndex - 1]?.isChecked;
+  const isCheckedRowAfter = tableRows.value[rowIndex]?.isChecked;
+  const isFirstRowAndNextChecked = rowIndex === 0 && isCheckedRowAfter;
+
+  (isPreviousRowChecked && isCheckedRowAfter) || isFirstRowAndNextChecked ? true : false;
```

Comment:
return? true and false may be removed. 

`return (isPreviousRowChecked && isCheckedRowAfter) || isFirstRowAndNextChecked;`